### PR TITLE
Orientjs upgrade and update to support orientjs v3.0.0 features .

### PR DESCRIPTION
**Pull request by saeedtabrizi**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by saeedtabrizi** [comment]
All new feature of orientjs has declared in this definition update . 
- cover issue #4785 now .
- adde some documentation and features .

File: orientjs/orientjs.d.ts [hunk 1]
```diff
@@ -44,67 +79,113 @@ declare module "orientjs" {
...
+
+            function Manager(config: MigrationManagerConfig): void;
             interface Migration {
                 name: string;
-                server: Server;
+                Server: Server;
                 db: Db;
                 configure(config?: any): void;
                 up(): Promise<any>;
                 down(): Promise<any>;
```
Comment (aozgaa) [inline orientjs/orientjs.d.ts]: nit: can you remove this newline as well?

Comment (saeedtabrizi) [inline orientjs/orientjs.d.ts]: @aozgaa , i correct to case of server ; Server is invalid and server is valid .

**Comment by saeedtabrizi** [comment]
Hi @aozgaa , I fixed some issues on `orientjs.d.ts` , `index.d.ts` . please merge this pull request .

**Comment by aozgaa** [comment]
Can you resolve the merge conflicts?

**Comment by saeedtabrizi** [comment]
@aozgaa I have no any acres to resolve it .

**Comment by zhengbli** [comment]
@saeedtabrizi can you merge the conflict on your branch? Otherwise there is no way to merge it. Thanks!

**Comment by saeedtabrizi** [comment]
@zhengbli I have no any conflict in my local branch and i have no any access to resolve because i have no write access to resolve this conflicts .

**Comment by zhengbli** [comment]
so if pull the latest `master` branch, and do `git merge master` locally, would you see conflicts?

**Comment by jramsay** [comment]
@saeedtabrizi: are you able to update this PR as per @zhengbli's instructions?

**Comment by saeedtabrizi** [comment]
@zhengbli No , i have no detect any conflict in my code .

**Comment by paulvanbrenk** [comment]
I tried to manually merge this, but it looks like your changes are based on a very old version of the definition file. Please consider manually re-applying the changes on a clean copy of master and submitting a new PR.

**Pull request closed without merging**